### PR TITLE
Handling arrays in Elgg API call

### PR DESCRIPTION
**Pull request by Peter83**
Handling arrays of elements,  when creating the API call and server side when receiving.

### CODE REVIEW

File: mod/web_services/lib/client.php [hunk 1]
```diff
@@ -158,3 +166,38 @@ function serialise_api_headers(array $headers) {
...
+            return $arr;
+        }
+
+        $r = array();
+        foreach ($arr as $k => $v) {
+            if (is_null($v)) {
+                continue;
+            }
+
+            if ($prefix && $k && !is_int($k)) {
```
Comment (Peter83) [inline mod/web_services/lib/client.php]: if ($prefix && isset($k)) {

should solve the issue when using "0" as array key

**Comment by jdalsem** [comment]
What exactly are you trying to add with this feature?

**Comment by Peter83** [comment]
> What exactly are you trying to add with this feature?

manage arrays as API input with a recursive function

**Comment by hypeJunction** [comment]
Didn't @mrclay already add that?

**Comment by jeabakker** [comment]
Because of a complete rewrite of the webservices plugin in #13121 we'll leave the current implementation as.

**Comment by jeabakker** [comment]
the issue with correctly encoding arrays is fixed in the rewrite

**Pull request closed without merging**